File: 75-sort-colors/75-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        low = mid = 0
        high = length - 1
        while(mid <= high):
            if(nums[mid] == 0):
                nums[low],nums[mid] = nums[mid], nums[low]
                low += 1
                mid += 1
            elif(nums[mid] == 1):
                mid += 1
            else:
                nums[mid],nums[high]= nums[high], nums[mid]
                high -= 1
        
        

File: 75-sort-colors/75-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int len = nums.length;
        int low = 0, mid = 0, high = len - 1,temp=0;
        while(mid <= high){
            if(nums[mid] == 0){
                temp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = temp;
                low++;
                mid++;
            } else if(nums[mid] == 1){
                mid++;
            } else {
                temp = nums[high];
                nums[high] = nums[mid];
                nums[mid] = temp;
                high--;
            }
        }
    }
}